src/bun.js/webcore/response.zig

```diff
@@ -828,27 +818,126 @@ pub const Fetch = struct {
+                        
+                        if (headers_.as(FetchHeaders)) |headers__| {
+                            headers = Headers.from(headers__, bun.default_allocator) catch unreachable;
+                            // TODO: make this one pass
+                        } else if (FetchHeaders.createFromJS(ctx.ptr(), headers_)) |headers__| {
+                            headers = Headers.from(headers__, bun.default_allocator) catch unreachable;
+                            headers__.deref();
+                        } else if (request.headers) |head| {
+                            headers = Headers.from(head, bun.default_allocator) catch unreachable;
+                        }
+                        
+                    } else if (request.headers) |head| {
+                        headers = Headers.from(head, bun.default_allocator) catch unreachable;
+                    }
 
-            url = ZigURL.parse(url_slice.slice());
+                    if (options.fastGet(ctx.ptr(), .body)) |body__| {
+                        if (Body.Value.fromJS(ctx.ptr(), body__)) |body_const| {
+                            var body_value = body_const;
+                            // TODO: buffer ReadableStream?
+                            // we have to explicitly check for InternalBlob
+                            body = body_value.useAsAnyBlob();
+                        } else {
+                            // an error was thrown
+                            return JSC.JSValue.jsUndefined().asObjectRef();
+                        }
+                    } else {
+                        body = request.body.useAsAnyBlob();
+                    }
 
+                    if (options.get(ctx, "timeout")) |timeout_value| {
+                        if (timeout_value.isBoolean()) {
+                            disable_timeout = !timeout_value.asBoolean();
+                        } else if (timeout_value.isNumber()) {
+                            disable_timeout = timeout_value.to(i32) == 0;
+                        }
+                    }
+
+                    if (options.get(ctx, "redirect")) |redirect_value| {
+                        if (redirect_value.getZigString(globalThis).eqlComptime("manual")) {
+                            follow_redirects = false;
+                        }
+                    }
+
+                    if (options.get(ctx, "keepalive")) |keepalive_value| {
+                        if (keepalive_value.isBoolean()) {
+                            disable_keepalive = !keepalive_value.asBoolean();
+                        } else if (keepalive_value.isNumber()) {
+                            disable_keepalive = keepalive_value.to(i32) == 0;
+                        }
+                    }
+                    if (options.get(globalThis, "verbose")) |verb| {
+                        verbose = verb.toBoolean();
+                    }
+                    if (options.get(globalThis, "proxy")) |proxy_arg| {
```

Comment:
We should:
- check if `proxy_arg` is undefined and skip this branch
- check if `proxy_arg` is null, in which case it should make the proxy null (i.e. let the user ignore their system proxy for a request)

Reply:
done